### DISCUSSION TITLE
PEPOLAR & fieldmap-less estimation of ME-EPI

### DISCUSSION THREAD

**Discussion by oesteban** [nipreps/fmriprep#2606] [General]
I wanted to spin this conversation off of #2530, because it is hard for me to come up with a suggestion.

Bottom line, the problem here is that each echo has a different `EffectiveEchoSpacing`/`TotalReadoutTime`. Please @tsalo @satra or @emdupre correct me if I'm wrong, but we expect the shorter echoes to have smaller distortions due to inhomogeneity of the B0 field than that for longer echoes.

In the PEPOLAR case, if the researcher makes use of the new `B0FieldIdentifier` field, they can explicitly define which echo is to be used with EPI images under fmap (or other EPIs under the func/ folder with different PE direction). Only the echoes selected for fieldmap estimation should be marked with `B0FieldIdentifier`.

The underlying problem is that, if we keep choosing the shortest echoes like we used to do for fMRIPrep, the distortion is oftentimes really hard to appreciate, and the registration (built-in in TOPUP or SyN) will not be sensitive to it and basically overfit in weird ways.

The general recommendation here (again, happy to stand corrected) should be to use the echo closest to the midpoint, where the trade-off between SNR and a distortion sufficiently present to make a decent estimation is best.

For the fieldmap-less option with ants' SyN, this choice is even more necessary because the shortest echo sometimes seems very mildly warped, and on the other hand, the contrast has proven to be very tricky now and in our older implementations of SyN, suggesting that less intensity in the WM and less signal from the fat of the head would be interesting to have.

Please let me know if the problem is sufficiently described.

**Comment by tsalo** [comment]
I'm not a physicist, so I don't actually know much about this, but I've asked more knowledgeable members of the `tedana` community about it in the past. The ones who would probably know best are @handwerkerd and @dowdlelt. 

Back in August, @dowdlelt did some comparisons on some of his multi-echo data and he couldn't find any evidence of differences in distortion across echoes. Sorry, I can't link to the actual conversation because it's on Gitter and apparently Gitter posts aren't linkable.

@handwerkerd followed up in that thread to say the following:

>Longer readouts definitely lead to more distortion & are what we usually focus on when talking about distortion. Longer echo times should be more sensitive to field inhomogeneties, T2* effects, etc. Those should also cause some spatial distortions, but not as much as a longer readout. We've ignored this with multi-echo both because it's not a huge effect & because we don't have a great way to deal with it. What I don't know is if it's a relatively small effect or an insignificantly small effect.

I don't know if that helps at all, but hopefully one of them can jump in on this conversation if there's more that needs to be added.

**Reply by oesteban** [reply]
thanks much for the prompt answer. I'm not sure I understand correctly. The question would be, probably, *is the effective echo spacing (as in, the time between reading two contiguous voxels along the PE direction) exactly the same for the N echos?*. If that were the case, then the susceptibility distortions should be the same between echos (which seems more intuitive as I haven't found anywhere where SDC is performed before calculating an optimally combined series).

Considering there's no preference as to which of the echos shows the distortion more presently, I guess centered echos should give a contrast more similar to a single-echo equivalent sequence.

Alternatively, contrast-wise again, wouldn't it make more sense to run PEPOLAR or SDC-SyN on the optimally combined series?

**Reply by dowdlelt** [reply]
My two cents: 

My understanding is that the effective echo spacing is identical - and the bulk susceptibility distortion is the same. As @handwerkerd points out, there are other effects. The largest being that dropout changes dramatically as TE increases and this can manifest as the appearance of increased distortion (for example, as the orbitofrontal cortex fades away over echo times) but actually isn't. The minimal distortion in early echoes (and all echoes) is likely due more to the in-plane acceleration often used with multiecho acquisition.

  Dan is correct that there are other effects as well (and of course feel free to correct me), but I have only seen something like this demonstrated in the vasculature in anatomical imaging (T2 GRE), here: https://figshare.com/articles/media/7T_in-vivo_human_T2_mapping_at_350_m_iso_res_using_ME-GRE_with_flow_artifact_mitigation_reveals_cortical_layers_vessels/13360100 and I haven't seen anything that makes me think it is a large enough effect in the context of fMRI to stress about. I would worry about getting things more wrong in an attempt to correct such small effects, as we really care about the the voxel by voxel information in multiecho. 

While not conclusive, in some 8 echo dataset I have sitting around, the structures visible in the first and last echoes do not appear to move, even as I shift between them. 
![image](https://user-images.githubusercontent.com/15126366/137927073-3fcdcc17-2962-439b-b5f0-17d376b15c3a.png)

Or this bright voxel in the image (in the eye, which tend to distort) which remains in a consistent location from echo 1 -> 4-> 8. (thats 9ms -> 46ms -> 96ms). 

![image](https://user-images.githubusercontent.com/15126366/137928068-0dc51359-5753-4a39-a7b4-54d4a849fdae.png)

While later echos could provide more contrast, it isn't clear to me that that would be the best choice - dropout means you can't correct what isn't present. I prefer the first echo (indeed, the SBRefs for the first echoes), as it has the most tissue to work with and provides the most information about the total brain shape. Correcting from a latter echo could easily misplace tissues that are present in earlier but not later echoes. 

The optimal combination is also a reasonable choice of course as it could marry the benefits of the first echo with better contrast, but I'm not sure about the pipeline and order of operations here. It also could get dicey as differences or errors in the calculation of T2star between forward and reverse phase encoding means that these images aren't as identical as your typical blip up/blip down pairs, again leading me to prefer the first echo approach. That said, the SyN approach seems like it may benefit much more from the optimal combination, as the first echo isn't quite what it was optimized for. One can also imagine calculating it for each echo and doing something clever to robustly 'average' those distortion estimates.

 Hope this is helpful.

**Reply by oesteban** [reply]
Thanks, this is very helpful - it does seem that the distortion is consistent across echos, with the associated dropout growing with echo time.

**Comment by oesteban** [comment]
Okay, except for some details I will soon update in the PR's thread, #2530 is "formally" (as in, the workflow builds and runs till the end) finished.

To test the multiecho pathways of the compute graph, I'm using ds000210, so it is probable some of the things I'm about to propose are not very representative of the average ME dataset in the wild.

Using ds000210 at its original resolution (we severely downsampled it for the automated tests, which basically changes everything), I'm finding that registration to the T1w image is very poor (and I would expect something similar to happen with a PEPOLAR approach, should the dataset contain EPI images with the opposed PE blips). Perhaps it's just ds000210 (with 3 echoes), but the first and second shortest echoes have the first volume with a contrast completely different to that of any subsequent timepoint. It is easier to see on the second echo, as the ventricles show clearly brighter than anything else and then become darker after that first frame (not sure right now whether my visualization tool is adjusting automatically the intensity, so I don't know if overall all intensities drop after the first or if roughly the whole brain is constant except the ventricles). The effect is also present on the shortest echo - which is the one we currently use.

So for the shortest echo, fMRIPrep standardizes intensity and identifies a number of volumes in the beginning as _nonsteady states_. We average the nonsteady states and use it as BOLD reference, which has worked well for SE images. Considering that the GM/WM contrast is not apparent on the shortest echo, with this additional effect of CSF "switching off" after the first volume, the result in the ME case is that we obtain a reference average that is almost constant throughout the brain. Without contrast, registration is really difficult.

Solutions I can propose at the moment:

- Use only the first volume of the shortest echo - for ds000210 this could work out very well, but I don't know whether this very different first volume is something expected from any other ME dataset, and how many volumes show this particular pattern. Since this seems pretty unreliable, I don't think I would go with this one.
- Do not use nonsteady-states for ME. We use nonsteady-states for SE because they have more T1w contrast and more WM/GM contrast overall. However, for ME it would make sense to actually skip the nonsteady states and average the following, say 5 to 10 volumes of the shortest echo. You get more contrast because the echo time is short but you don't capture the initial instability of the scanner.
- Use the second-shortest echo. fMRIPrep considers ME when the dataset has 3 echoes or more. Therefore, the second shortest echo will usually fall around the echo time of a standard SE sequence. And in general, our strategy has worked fine for SE datasets. This solution will ignore the fact again that the first (or maybe more or absent in other datasets) may have an extremely different contrast.

Of course, we could mix and match solutions. E.g., (i) do not pick nonsteady states for averaging; and (ii) average the first 10 volumes after steady state is reached, first averaging the shortest two echoes volume-wise (which could be more sophisticated than just an average, to account for excessive dropout) and then averaging those through time.

WDY'allT? I'm in particular interested in @tsalo's views because I think he has actually faced this kind of problem before.

**Reply by tsalo** [reply]
I like the third idea- to use the standard SE approach with the second-shortest echo. I think it would probably perform well and would be the most consistent with fMRIPrep's SE workflow. The only change I'd recommend would be to identify the echo closest to the standard echo time based on the scanner strength- so 30ms for 3T. This might be different from the second echo for ME protocols with four or more echoes. They aren't as common as three-echo protocols, but they definitely occur.

As echo time increases, we should see contrast also increase, but coverage should decrease. Would increased coverage be relevant here? E.g., would it be better to run a volume-wise T2* estimation workflow and optimal combination just on the non-steady-state volumes to produce un-MCed combined time series that maximize both contrast and coverage, which could then be used to produce the reference image?

**Reply by oesteban** [reply]
> I like the third idea- to use the standard SE approach with the second-shortest echo. I think it would probably perform well and would be the most consistent with fMRIPrep's SE workflow.

Cool, good to see I wasn't missing the point by a whole lot this time :)

It's also good that we can quickly try the new implementation and then incorporate a smarter echo-selection strategy down the line.



> As echo time increases, we should see contrast also increase, but coverage should decrease. Would increased coverage be relevant here?

Yes, a lot. This is why I propose this idea in terms of "second-shortest" echo rather than the echo closest to a standard SE equivalent. With 3 echos you probably want to avoid the last echo, even in the weird case (but possible) that it turns out to be closest to 30 ms (3T). With four or more, picking the second or the third will not be that big of a difference contrast-wise but I agree coverage may be very different.

The idea of the T2* map with only nonsteady states is something to consider, definitely.

Any voices against taking the very first step (i.e., stop selecting the shortest echo and use the second shortest echo instead for the reference)?

**Reply by eilidhmacnicol** [reply]
If I may contribute from my narrow experience:

WRT the choice of echo, I would definitely back if (one of) the early echoes are selected rather than based on proximity to a lookup TE value. To give an example, our 4 echo protocols previously used echoes at 11->19->27->35 ms (although we have been using 6->16->26->36 ms more recently). 

<img width="647" alt="Screenshot 2021-10-26 at 17 22 24" src="https://user-images.githubusercontent.com/54267723/138920460-b96ed37b-378e-48f7-9537-6fd4b744c02c.png">

> With 3 echos you probably want to avoid the last echo, even in the weird case (but possible) that it turns out to be closest to 30 ms (3T). With four or more, picking the second or the third will not be that big of a difference contrast-wise but I agree coverage may be very different.

The first and second echo are equidistant from a standard TE of 15 ms at 9.4 T, so some secondary heuristic would be needed to decide between the two echoes. However, even only 4ms after the standard TE, there is already a large drop in signal intensity, particularly in the dorsal skull and ventral brain surface. It's been stated above that distortions cannot be corrected if it there is no signal. So, if two values are equally close to a standard value, would the value that is closest but before the standard TE be used to calculate distortions?

All this to say, I think choosing the second echo would mitigate both problems of poor signal and poor contrast (at least for images with 3-4 echoes, but I suppose that depends on the number of echoes available). I don't know enough to suggest how it would be done, but it would seem that if distortions could be calculated from the optimally combined would provide an impartial way forward that would span datasets with minimal decisions impacting the decision.

>  I don't know whether this very different first volume is something expected from any other ME dataset

I haven't seen this phenomenon in our data so I'd go out on a limb and say that I don't think it is expected. I would guess it's because we don't acquire our dummy scans (the volumes are discarded online during reconstruction), which also means that we're less likely to have non-steady state volumes to create the T2* map with, but sounds like that's something down the line.

**Reply by oesteban** [reply]
Okay, if I understand correctly, then @eilidhmacnicol would lean towards keep using the shortest echo, but instead of using the nonsteady states (which are removed in reconstruction, for their settings), use a number of timepoints not too late in the run, but when the scanner is in its stable regime, correct?

**Reply by handwerkerd** [reply]
Sorry for a delayed response on this thread. Thinking ahead to when [EPTI](https://pubmed.ncbi.nlm.nih.gov/30714198/) sequences become more common, and echoes are much closer to each other, just deciding to use the second echo would lead to problems.

The core issue seems to be that the earliest echo has the least dropout, but little gray/white contrast. (FWIW, gray/white contrast should increase slowly with echo time as dropout increases, but the CSF/gray contrast will increase a lot faster. I think it's the CSF/GM difference that might give clearer structure to a brain.)

A simple metric, as proposed above, would be to take estimated optimal T2* time for a field strength (i.e. around 28-30ms for 3T) and use the echo that closest, but below that value. This will almost always be the second echo for 3-4 echo time series, but may be a later echo if someone pushes in-slice acceleration to get >4 echoes.
A fancier method would be to estimate dropout & contrast and set some threshold for when the contrast improvement is better than the dropout loss. This is probably not worth the effort when most people are using 3-5 echoes, but may be something to consider in the future.
